aten/src/ATen/native/ReduceOps.cpp

```diff
@@ -349,12 +349,16 @@ Tensor _prod(const Tensor &self, int64_t dim_, bool keepdim) {
 Tensor& logsumexp_out(Tensor& result, const Tensor &self, int64_t dim_, bool keepdim) {
   int64_t dim = maybe_wrap_dim(dim_, self.dim());
   // can't take max of empty tensor
-  AT_CHECK(self.numel() != 0, "logsumexp only works on nonempty tensors");
-  auto maxes = at::max_values(self, dim, true);
-  auto maxes_squeezed = (keepdim ? maxes : maxes.squeeze(dim));
-  maxes_squeezed.masked_fill_(maxes_squeezed.abs() == INFINITY, 0);
-  at::sum_out(result, at::exp(self - maxes), dim, keepdim);
-  result.log_().add_(maxes_squeezed);
+  if (self.numel() != 0) {
+    auto maxes = at::max_values(self, dim, true);
+    auto maxes_squeezed = (keepdim ? maxes : maxes.squeeze(dim));
+    maxes_squeezed.masked_fill_(maxes_squeezed.abs() == INFINITY, 0);
+    at::sum_out(result, at::exp(self - maxes), dim, keepdim);
+    result.log_().add_(maxes_squeezed);
+  } else {
+    at::sum_out(result, at::exp(self), dim, keepdim);
+    result.log_();
+  }
```

Comment:
I'm a little concerned because `scipy.misc.logsumexp(np.ndarray([])) = -2.0`, while our implementation gives 
```
In [13]: torch.tensor([]).exp().sum().log()
Out[13]: tensor(-inf)
```
but that can be another issue

Reply:
Interesting.  It's appears to be a little complicated.

The equivalent operations in `np` also give `-inf`:
```
In [10]: np.log(np.sum(np.exp(np.array([]))))
Out[10]: -inf
```

`scipy.misc.logsumexp` docs suggest to use `scipy.special.logsumexp` instead which for my version gives an error:
```
scipy.special.logsumexp(np.array([]))
ValueError: zero-size array to reduction operation maximum which has no identity
```

Reply:
So I guess, I don't know where the -2.0 would come from...

Reply:
I think the -inf is correct (because it is the neutral element w.r.t. logsumexp).